pkg/deploy/controller/deployment/factory.go

```diff
@@ -122,6 +126,15 @@ func (factory *DeploymentControllerFactory) Create() controller.RunnableControll
 	}
 }
 
+// setPodAttributes updates the deployer pod attributes based on the deployment strategy configuration
+func (factory *DeploymentControllerFactory) setPodAttributes(strategy *deployapi.DeploymentStrategy, pod *kapi.Pod) {
+	switch strategy.Type {
+	case deployapi.DeploymentStrategyTypeCustom:
+		util.MergeInto(pod.Labels, strategy.Labels, 0)
```

Comment:
Is there a constant for 0 in this case?  If so, we should use it, if not,
we should add it (so the behavior is obvious.

On Oct 31, 2015, at 3:24 PM, Jordan Liggitt <notifications@github.com>
wrote:

In pkg/deploy/controller/deployment/factory.go
<https://github.com/openshift/origin/pull/5149#discussion_r43575255>:

> @@ -122,6 +126,15 @@ func (factory *DeploymentControllerFactory) Create() controller.RunnableControll
>  	}
>  }
>
> +// setPodAttributes updates the deployer pod attributes based on the deployment strategy configuration
> +func (factory *DeploymentControllerFactory) setPodAttributes(strategy *deployapi.DeploymentStrategy, pod *kapi.Pod) {
> +	switch strategy.Type {
> +	case deployapi.DeploymentStrategyTypeCustom:
> +		util.MergeInto(pod.Labels, strategy.Labels, 0)

Got it, hadn't checked the MergeInto impl. Add a comment here noting that
we're not overwriting existing labels on the pod?

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/5149/files#r43575255>.